nixos/doc/manual/configuration/luks-file-systems.xml

```diff
@@ -37,4 +37,37 @@ Enter passphrase for /dev/disk/by-uuid/3f6b0024-3a44-4fde-a43a-767b872abe5d: ***
   on an encrypted partition, it is necessary to add the following grub option:
 <programlisting><xref linkend="opt-boot.loader.grub.enableCryptodisk"/> = true;</programlisting>
  </para>
+  <section xml:id="sec-luks-file-systems-fido2">
+  <title>FIDO2</title>
+
+  <para>
+   NixOS also supports unlocking your LUKS-Encrypted file system using an FIDO2 compatibly token. In the following example, we create a new FIDO2 credential
+   and add it as a new key to our existing device <filename>/dev/sda2</filename>:
+
+   <screen>
+# fido2luks credential $HOSTNAME
+f1d00200108b9d6e849a8b388da457688e3dd653b4e53770012d8f28e5d3b269865038c346802f36f3da7278b13ad6a3bb6a1452e24ebeeaa24ba40eef559b1b287d2a2f80b7
+
+# fido2luks -i add-key /dev/sda2 f1d00200108b9d6e849a8b388da457688e3dd653b4e53770012d8f28e5d3b269865038c346802f36f3da7278b13ad6a3bb6a1452e24ebeeaa24ba40eef559b1b287d2a2f80b7
+Password:
+Password (again):
+Old password:
+Old password (again):
+Added to key to device /dev/sda2, slot: 2
+</screen>
+
+  To ensure that this file system is automatically decrypted using the FIDO2 compatible key, add the following to <filename>configuration.nix</filename>:
```

Comment:
This is a great PR, thank you!  Question -- does this require tapping the physical token?

Reply:
@grahamc I did test it with Trezor and it requires a confirmation.

Reply:
Oh, I see what you mean, probably using _automatically_ doesn't make sense here.

Reply:
:grin:

Reply:
Looking forward to using this PR and also one-day having my secure-boot PR merged.